src/mscorlib/shared/System/Version.cs

```diff
@@ -235,18 +237,15 @@ public bool TryFormat(Span<char> destination, int fieldCount, out int charsWritt
 
         private StringBuilder ToCachedStringBuilder(int fieldCount)
         {
-            if (fieldCount == 1)
-            {
-                StringBuilder sb = StringBuilderCache.Acquire();
-                AppendNonNegativeNumber(_Major, sb);
-                return sb;
-            }
-            else if (fieldCount == 2)
+            // Note: As we always have positive numbers then it is safe to convert the number to string
+            // regardless of the current culture as we'll not have any punctuation marks in the number.
+
+            if (fieldCount == 2)
             {
                 StringBuilder sb = StringBuilderCache.Acquire();
-                AppendNonNegativeNumber(_Major, sb);
+                sb.Append(_Major);
                 sb.Append('.');
-                AppendNonNegativeNumber(_Minor, sb);
+                sb.Append(_Minor);
```

Comment:
I expect this is slightly slower, in particular as it'll end up looking up the current culture from TLS.  But it should only be a few percent, this shouldn't be a particularly hot path, and it's nice to be able to get rid of the custom formatting logic.